.github/workflows/integration.yml

```diff
@@ -69,25 +72,25 @@ jobs:
       run: time ./gradlew clean checkstyleMain checkstyleTest test systemTest coverage
 
     - name: Run code coverage
-      if: ${{ success() }}
+      if: ${{ success() && matrix.os == 'ubuntu-20.04' }}
```

Comment:
the `&& matrix.os == 'ubuntu-20.04'` feels a little repetitive, not sure if you can like `if` this once run all these jobs.

Reply:
I moved it out to a separate job that will only run if the original build job was successful.

Reply:
Do you think that something like matrix.os[last index] should be used instead of 'ubuntu-20.04'?

Reply:
It is possible, but I think it might be a bit too complicated, because we need to count the number of entries in the build matrix first to determine the index of the last OS. When updating the build matrix in the future, it would be easier to search for occurrences of the string `ubuntu-20.04`.

Reply:
If not the last element, maybe the first element is good?

Reply:
I think we shouldn't run it on the first element which is not Ubuntu 20.04. This way of specifying ensures that we know the exact OS version that this step is running on.